addons/sale_stock/models/sale_order.py

```diff
@@ -485,6 +485,16 @@ def _onchange_product_uom_qty(self):
             return {'warning': warning_mess}
         return {}
 
+    def action_product_forecast_report(self):
+        self.ensure_one()
+        action = self.product_id.action_product_forecast_report()
+        action['context'] = {
+            'active_id': self.product_id.id,
+            'active_model': 'product.product',
+            'order_name': self.order_id.name,
```

Comment:
The name is not mandatory unique (even it should be in the most situation, but there isn't a hard constraint about it). Maybe pass two values, model and id, is more elegant and correct in any situation.
E.g. `'source_document_model' :  'sale.order', 'source_document_id': self.order_id.id`
And use these values later on the forecasted.

Reply:
I'm not sure I understand well. Do you mean that `name`is not the best to do the match in forecasted? (in that case I guess it applies to PO and MO as well).
I could pass some `move_id` so that I can do the match directly in forecasted in a generic way.

Reply:
"Do you mean that nameis not the best to do the match in forecasted?", Yes.Because the name is not unique in all case, you should avoid use it to match lines. 
"I could pass some move_id so that I can do the match directly in forecasted in a generic way.": it seems perfect (better than model:id solution).